classification/keras/code/keras_script.py

```diff
@@ -0,0 +1,482 @@
+    standardize=custom_standardization,
+    max_tokens=data_params["max_features"],
+    output_mode="int",
+    output_sequence_length=data_params["sequence_length"],
+)
+
+text_ds = raw_train_ds.map(lambda x, y: x)
+vectorize_layer.adapt(text_ds)
+
+
+# ### Vectorize data
+
+# In[51]:
+
+
+def vectorize_text(text, label):
+    text = tf.expand_dims(text, -1)
+    return vectorize_layer(text), label
+
+
+# Vectorize the data.
+train_ds = raw_train_ds.map(vectorize_text)
+val_ds = raw_val_ds.map(vectorize_text)
+test_ds = raw_test_ds.map(vectorize_text)
+
+# Do async prefetching / buffering of the data for best performance on GPU.
+train_ds = train_ds.cache().prefetch(buffer_size=10)
+val_ds = val_ds.cache().prefetch(buffer_size=10)
+test_ds = test_ds.cache().prefetch(buffer_size=10)
+
+
+# ## Modelling
+
+# (Neptune) Create a new model and model version
+
+# In[52]:
+
+
+from neptune.new.exceptions import NeptuneModelKeyAlreadyExistsError
+
+project_key = project.get_structure()["sys"]["id"].fetch()
+
+try:
+    model = neptune.init_model(name="keras", key="KER")
+    model.stop()
+    model_version = neptune.init_model_version(model=f"{project_key}-KER", name="keras")
+except NeptuneModelKeyAlreadyExistsError:
+    model_version = neptune.init_model_version(model=f"{project_key}-KER", name="keras")
+
+
+# ### Build a model
+
+# In[53]:
+
+
+model_params = {
+    "dropout": 0.5,
+    "strides": 3,
+    "activation": "relu",
+    "kernel_size": 7,
+    "loss": "binary_crossentropy",
+    "optimizer": "adam",
+    "metrics": ["accuracy"],
+}
+
+
+# In[54]:
+
+
+model_version["params"] = model_params
+
+
+# In[55]:
+
+
+from tensorflow.keras import layers
+
+# A integer input for vocab indices.
+inputs = tf.keras.Input(shape=(None,), dtype="int64")
+
+# Next, we add a layer to map those vocab indices into a space of dimensionality
+# 'embedding_dim'.
+x = layers.Embedding(data_params["max_features"], data_params["embedding_dim"])(inputs)
+x = layers.Dropout(model_params["dropout"])(x)
+
+# Conv1D + global max pooling
+x = layers.Conv1D(
+    data_params["embedding_dim"],
+    model_params["kernel_size"],
+    padding="valid",
+    activation=model_params["activation"],
+    strides=model_params["strides"],
+)(x)
+x = layers.Conv1D(
+    data_params["embedding_dim"],
+    model_params["kernel_size"],
+    padding="valid",
+    activation=model_params["activation"],
+    strides=model_params["strides"],
+)(x)
+x = layers.GlobalMaxPooling1D()(x)
+
+# We add a vanilla hidden layer:
+x = layers.Dense(data_params["embedding_dim"], activation=model_params["activation"])(x)
+x = layers.Dropout(model_params["dropout"])(x)
+
+# We project onto a single unit output layer, and squash it with a sigmoid:
+predictions = layers.Dense(1, activation="sigmoid", name="predictions")(x)
+
+keras_model = tf.keras.Model(inputs, predictions)
+
+# Compile the model with binary crossentropy loss and an adam optimizer.
+keras_model.compile(
```

Comment:
As nothing Neptune related happens, would be nice to move this model creation code to a function in `utils`.